library/alloc/src/boxed.rs

```diff
@@ -543,7 +543,7 @@ impl<T, A: Allocator> Box<T, A> {
     #[inline(always)]
     pub const fn pin_in(x: T, alloc: A) -> Pin<Self>
     where
-        A: 'static + ~const Allocator + ~const Drop,
+        A: PinSafeAllocator + ~const Allocator + ~const PinSafeAllocator + ~const Drop,
```

Comment:
I don't think you need both of these bounds, just the `~const PinSafeAllocator` one. `A: ~const PinSafeAllocator` implies `A: PinSafeAllocator`.

Reply:
Ah, you're right. I'm new to the `~const Trait` syntax. This is fixed now.